### PR TITLE
Use pypechain package

### PR DESCRIPTION
**Pull request by sentilesdal**
Removes the local pypechain lib in favor of pypi package.  The types in hypertypes is now regenerated with the latest types.

- remove pypechain folder
- remove lateral dependencies
- update pypechain dependency.  remove script.
- fix codegen test
- update types
- update README.md

### CODE REVIEW

File: lib/hypertypes/hypertypes/ERC4626HyperdriveDeployerContract.py [hunk 1]
```diff
@@ -45,7 +52,13 @@ def __init__(self, address: ChecksumAddress | None = None, abi=Any) -> None:
...
         # TODO: make this better, shouldn't initialize to the zero address, but the Contract's init
         # function requires an address.
-        self.address = address if address else cast(ChecksumAddress, "0x0000000000000000000000000000000000000000")
+        self.address = (
+            address
+            if address
+            else cast(
+                ChecksumAddress, "0x0000000000000000000000000000000000000000"
+            )
+        )
```
Comment (wakamex) [inline lib/hypertypes/hypertypes/ERC4626HyperdriveDeployerContract.py:61]:
```suggestion
        self.address = (address or cast(
                ChecksumAddress, "0x0000000000000000000000000000000000000000"
            )
        )
```

Comment (wakamex) [reply lib/hypertypes/hypertypes/ERC4626HyperdriveDeployerContract.py:61]: this would also replace "falsy" values like an empty string 🤔

Comment (wakamex) [reply lib/hypertypes/hypertypes/ERC4626HyperdriveDeployerContract.py:61]: wait, this is machine generated isn't it

Comment (sentilesdal) [reply lib/hypertypes/hypertypes/ERC4626HyperdriveDeployerContract.py]: yeah it is, but we can change the generated code.  would be a suggestion for pypechain though, which i can do.

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 2]
```diff
@@ -29,7 +29,15 @@ def local_chain() -> Iterator[str]:
     # Using context manager here seems to make CI hang, so explicitly killing process at the end of yield
     # pylint: disable=consider-using-with
     anvil_process = subprocess.Popen(
-        ["anvil", "--host", "127.0.0.1", "--port", str(anvil_port), "--code-size-limit", "9999999999"]
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:32]: 80 char gutter?

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 3]
```diff
@@ -44,7 +52,9 @@ def local_chain() -> Iterator[str]:
 
 
 @pytest.fixture(scope="function")
-def local_hyperdrive_pool(local_chain: str) -> DeployedHyperdrivePool:  # pylint: disable=redefined-outer-name
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:47]: 80 char gutter?

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 4]
```diff
@@ -73,7 +83,9 @@ def local_hyperdrive_pool(local_chain: str) -> DeployedHyperdrivePool:  # pylint
     # Lots of local  variables for the tests
     # pylint: disable=too-many-locals
     # Deployer is the pre-funded account on the Delv devnet
-    deployer_private_key: str = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+    deployer_private_key: str = (
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:86]: 80 char gutter?

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 5]
```diff
@@ -85,8 +97,14 @@ def local_hyperdrive_pool(local_chain: str) -> DeployedHyperdrivePool:  # pylint
     max_curve_fee = FixedPoint("0.3")  # 30%
     max_flat_fee = FixedPoint("0.0015")  # 0.15%
     max_governance_fee = FixedPoint("0.30")  # 30%
-    fees = Fees(curve_fee.scaled_value, flat_fee.scaled_value, governance_fee.scaled_value)
-    max_fees = Fees(max_curve_fee.scaled_value, max_flat_fee.scaled_value, max_governance_fee.scaled_value)
+    fees = Fees(
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:100]: 80 char gutter?

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 6]
```diff
@@ -96,7 +114,8 @@ def local_hyperdrive_pool(local_chain: str) -> DeployedHyperdrivePool:  # pylint
     position_duration = 604800  # 1 week
     checkpoint_duration = 3600  # 1 hour
     time_stretch = FixedPoint(1) / (
-        FixedPoint("5.24592") / (FixedPoint("0.04665") * (initial_fixed_rate * FixedPoint(100)))
+        FixedPoint("5.24592")
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:117]: 80 char gutter?

File: lib/hypertypes/README.md [hunk 7]
```diff
@@ -3,9 +3,11 @@
 Generate the requisite Python files by running:
 
 ```bash
-sh scripts/build_type_files.sh path/to/ABIs num_chars_per_line
+sh pypechain path/to/ABIs --output_dir path/to/temp_dir
 ```
 
+and then moving the generated files to lib/hypertypes/hypertypes
```
Comment (dpaiton) [inline lib/hypertypes/README.md:9]: should it be "and then move ..."?

File: lib/ethpy/pyproject.toml [hunk 8]
```diff
@@ -42,4 +40,7 @@ lateral = [
 
 [build-system]
 requires = ["flit_core>=3.2"]
-build-backend = "flit_core.buildapi"
\ No newline at end of file
+build-backend = "flit_core.buildapi"
+
+[tool.black]
```
Comment (dpaiton) [inline lib/ethpy/pyproject.toml:45]: black is not an ethpy dependency, which is why we did not specify these tool configs in the individual subpackages. I think you should be able to delete lines 45 & 46.

File: lib/ethpy/ethpy/test_fixtures/local_chain.py [hunk 9]
```diff
@@ -86,7 +97,11 @@ def local_hyperdrive_pool(local_chain: str) -> DeployedHyperdrivePool:  # pylint
     max_flat_fee = FixedPoint("0.0015")  # 0.15%
     max_governance_fee = FixedPoint("0.30")  # 30%
     fees = Fees(curve_fee.scaled_value, flat_fee.scaled_value, governance_fee.scaled_value)
-    max_fees = Fees(max_curve_fee.scaled_value, max_flat_fee.scaled_value, max_governance_fee.scaled_value)
+    max_fees = Fees(
```
Comment (dpaiton) [inline lib/ethpy/ethpy/test_fixtures/local_chain.py:100]: 80 char gutter

File: lib/hypertypes/hypertypes/ERC20MintableContract.py [hunk 10]
```diff
@@ -66,11 +66,11 @@ class ERC20MintableBurnContractFunction(ContractFunction):
 
     def __call__(self, amount: int, destination: str | None = None) -> "ERC20MintableBurnContractFunction":
         if all([destination is None]):
-            super().__call__(amount)
+            super().__call__()
```
Comment (dpaiton) [inline lib/hypertypes/hypertypes/ERC20MintableContract.py:69]: these two conditions are the same now, so why have the conditional at all?

Comment (sentilesdal) [reply lib/hypertypes/hypertypes/ERC20MintableContract.py:69]: this actually looks like a bug

File: lib/hypertypes/README.md (hunk 7 above)
Comment (sentilesdal) [reply lib/hypertypes/README.md:9]: The complete sentence sounds right to me:

Generate the requisite Python files by running:

```bash
sh pypechain path/to/ABIs --output_dir path/to/temp_dir
```

and then moving the generated files to lib/hypertypes/hypertypes

File: lib/ethpy/pyproject.toml (hunk 8 above)
Comment (sentilesdal) [reply lib/ethpy/pyproject.toml]: removed

**Review by dpaiton — APPROVED** [review]

**Pull request merged**